### PR TITLE
zh-blog: v1.30 mid cycle preview of key enhancements

### PR DESCRIPTION
**Pull request by pacoxu**
Chinese version for https://kubernetes.io/blog/2024/03/12/kubernetes-1-30-upcoming-changes/

### CODE REVIEW

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+maintaining the integrity, security, and efficiency of cluster operations, making Kubernetes
+environments more robust and adaptable to various use cases and requirements. For more information
+on using CEL for admission control, see the [API
+documentation](/docs/reference/access-authn-authz/validating-admission-policy/) for
+ValidatingAdmissionPolicy.
+-->
+CEL 的加入使 Kubernetes 的准入控制赋予了集群管理员制定复杂规则的能力，
+这些规则可以根据集群的期望状态和策略来评估 API 请求的内容，而无需使用基于 Webhook 的访问控制器。
+这种控制水平对于维护集群操作的完整性、安全性和效率至关重要，使 Kubernetes 环境更加健壮，更适应各种用例和需求。
+有关使用 CEL 进行准入控制的更多信息，请参阅[API 文档](/zh-cn/docs/reference/access-authn-authz/validating-admission-policy/)中的 ValidatingAdmissionPolicy。
```
[the remark this answers is not in the document]
Comment (pacoxu) [reply content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md]: Updated.

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+<!--
+### Container resource based pod autoscaling ([KEP-1610](https://kep.k8s.io/1610))
+-->
+### 基于容器资源指标的 Pod 自动扩缩容 ([KEP-1610](https://kep.k8s.io/1610))
+
+<!--
+Horizontal pod autoscaling based on `ContainerResource` metrics will graduate to stable in v1.30.
+This new behavior for HorizontalPodAutoscaler allows you to configure automatic scaling based on the
+resource usage for individual containers, rather than the aggregate resource use over a Pod. See our
+[previous article](2023/05/02/hpa-container-resource-metric/) for further details, or read
```
[the remark this answers is not in the document]
Comment (pacoxu) [reply content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md]: Updated. 👍

**Review by ydFu — COMMENTED** [review]
/lgtm

thanks!

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+另一个重大变化是另一种模式：`LimitedSwap`。
+在`LimitedSwap`模式下，kubelet 实际上使用了该节点上的页面文件，并允许 Pod 的一些虚拟内存被分页出去。
+容器（及其父 Pod）在超出其内存限制的交换空间上没有访问权限，但系统仍然可以使用额外可用的交换空间。
+
+<!--
+Kubernetes' Node special interest group (SIG Node) will also update the documentation to help you
+understand how to use the revised implementation, based on feedback from end users, contributors,
+and the wider Kubernetes community.
+-->
+Kubernetes 的节点特别兴趣小组（SIG Node）还将根据最终用户、贡献者和更广泛的 Kubernetes 社区的反馈更新文档，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:126]:
```suggestion
Kubernetes 的 SIG Node 小组还将根据最终用户、贡献者和更广泛的 Kubernetes 社区的反馈更新文档，
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+以帮助你了解如何使用经过修订的实现。
+
+<!--
+Read the previous [blog post](/blog/2023/08/24/swap-linux-beta/) or the [node swap
+documentation](/docs/concepts/architecture/nodes/#swap-memory) for more details on 
+Linux node swap support in Kubernetes.
+-->
+阅读之前的[博客文章](/zh-cn/blog/2023/08/24/swap-linux-beta/)或[交换内存管理文档](/zh-cn/docs/concepts/architecture/nodes/#swap-memory)以获取有关 Kubernetes 中 Linux 节点交换支持的更多详细信息。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:134]:
```suggestion
阅读之前的[博客文章](/zh-cn/blog/2023/08/24/swap-linux-beta/)或[交换内存管理文档](/zh-cn/docs/concepts/architecture/nodes/#swap-memory)以获取有关
Kubernetes 中 Linux 节点交换支持的更多详细信息。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+### 支持 Pod 运行在用户命名空间 ([KEP-127](https://kep.k8s.io/127))
+
+<!--
+[User namespaces](/docs/concepts/workloads/pods/user-namespaces) is a Linux-only feature that better
+isolates pods to prevent or mitigate several CVEs rated high/critical, including
+[CVE-2024-21626](https://github.com/opencontainers/runc/security/advisories/GHSA-xr7r-f8xq-vfvv),
+published in January 2024. In Kubernetes 1.30, support for user namespaces is migrating to beta and
+now supports pods with and without volumes, custom UID/GID ranges, and more!
+-->
+[用户命名空间](/zh-cn/docs/concepts/workloads/pods/user-namespaces) 是一个仅在 Linux 上可用的功能，它更好地隔离了 Pod，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:148]:
```suggestion
[用户命名空间](/zh-cn/docs/concepts/workloads/pods/user-namespaces) 是一个仅在 Linux 上可用的特性，它更好地隔离 Pod，
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+以防止或减轻几个高/严重级别的 CVE，包括 2024 年 1 月发布的 [CVE-2024-21626](https://github.com/opencontainers/runc/security/advisories/GHSA-xr7r-f8xq-vfvv)。
+在 Kubernetes 1.30 中，对用户命名空间的支持正在迁移到 beta，并且现在支持带有和不带有卷的 Pod，自定义 UID/GID 范围等等！
+
+<!--
+### Structured authorization configuration ([KEP-3221](https://kep.k8s.io/3221))
+-->
+### API 服务器的结构化授权配置 ([KEP-3221](https://kep.k8s.io/3221))
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:155]:
```suggestion
### 结构化鉴权配置 ([KEP-3221](https://kep.k8s.io/3221))
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+configuration](/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file)
+is moving to beta and will be enabled by default. This feature enables the creation of
+authorization chains with multiple webhooks with well-defined parameters that validate requests in a
+particular order and allows fine-grained control – such as explicit Deny on failures. The
+configuration file approach even allows you to specify [CEL](/docs/reference/using-api/cel/) rules
+to pre-filter requests before they are dispatched to webhooks, helping you to prevent unnecessary
+invocations. The API server also automatically reloads the authorizer chain when the configuration
+file is modified.
+-->
+对[结构化授权配置](/zh-cn/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file)的支持正在迁移到 beta，并且将默认启用。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:168]:
```suggestion
对[结构化鉴权配置](/zh-cn/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file)的支持正在晋级到 Beta 版本，并将默认启用。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+这个特性使得可以创建具有多个明确定义参数的 Webhook 的授权链，按特定顺序验证请求，并允许进行细粒度的控制，例如在失败时明确拒绝。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:169]:
```suggestion
这个特性支持创建具有明确参数定义的多个 Webhook 所构成的鉴权链；这些 Webhook 按特定顺序验证请求，
并允许进行细粒度的控制，例如在失败时明确拒绝。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+配置文件方法甚至允许你指定 [CEL](/zh-cn/docs/reference/using-api/cel/) 规则，以在将请求分派到 Webhook 之前对其进行预过滤，帮助你防止不必要的调用。
+当配置文件被修改时，API 服务器还会自动重新加载授权链。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:171]:
```suggestion
当配置文件被修改时，API 服务器还会自动重新加载鉴权链。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+configuration file, those will continue to work as-is. To gain access to new authorization webhook
+capabilities like multiple webhooks, failure policy, and pre-filter rules, switch to putting options
+in an `--authorization-config` file. From Kubernetes 1.30, the configuration file format is
+beta-level, and only requires specifying `--authorization-config` since the feature gate is enabled by
+default. An example configuration with all possible values is provided in the [Authorization
+docs](/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file).
+For more details, read the [Authorization
+docs](/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file).
+-->
+你必须使用 `--authorization-config` 命令行参数指定授权配置的路径。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:185]:
```suggestion
你必须使用 `--authorization-config` 命令行参数指定鉴权配置的路径。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+如果你想继续使用命令行标志而不是配置文件，那么它们将继续按原样工作。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:186]:
```suggestion
如果你想继续使用命令行标志而不是配置文件，命令行方式没有变化。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+要获得新的授权 Webhook 功能，例如多个 Webhook、失败策略和预过滤规则，切换到将选项放在 `--authorization-config` 文件中。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:187]:
```suggestion
要访问新的 Webhook 功能，例如多 Webhook 支持、失败策略和预过滤规则，需要切换到将选项放在 `--authorization-config` 文件中。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 13]
```diff
@@ -0,0 +1,242 @@
...
+从 Kubernetes 1.30 开始，配置文件格式约定是 beta 级别的，只需要指定 `--authorization-config`，因为特性门控默认启用。
+[授权文档](/zh-cn/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file)中提供了一个包含所有可能值的示例配置。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:189]:
```suggestion
[鉴权文档](/zh-cn/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file)
中提供了一个包含所有可能值的示例配置。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 14]
```diff
@@ -0,0 +1,242 @@
...
+有关更多详细信息，请阅读[授权文档](/zh-cn/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file)。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:190]:
```suggestion
有关更多详细信息，请阅读[鉴权文档](/zh-cn/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file)。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 15]
```diff
@@ -0,0 +1,242 @@
...
+
+<!--
+Horizontal pod autoscaling based on `ContainerResource` metrics will graduate to stable in v1.30.
+This new behavior for HorizontalPodAutoscaler allows you to configure automatic scaling based on the
+resource usage for individual containers, rather than the aggregate resource use over a Pod. See our
+[previous article](/blog/2023/05/02/hpa-container-resource-metric/) for further details, or read
+[container resource metrics](/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics).
+-->
+基于 `ContainerResource` 指标的 Pod 水平自动扩缩容将在 v1.30 中升级为稳定版。
+这种 HorizontalPodAutoscaler 的新行为允许你根据单个容器的资源使用情况而不是 Pod 的聚合资源使用情况来配置自动伸缩。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:205]:
```suggestion
HorizontalPodAutoscaler 的这一新行为允许你根据各个容器的资源使用情况而不是 Pod 的聚合资源使用情况来配置自动伸缩。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 16]
```diff
@@ -0,0 +1,242 @@
...
+有关更多详细信息，请参阅我们的[先前文章](/zh-cn/blog/2023/05/02/hpa-container-resource-metric/)，或阅读[容器资源指标](/zh-cn/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics)。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:206]:
```suggestion
有关更多详细信息，请参阅我们的[先前文章](/zh-cn/blog/2023/05/02/hpa-container-resource-metric/)，
或阅读[容器资源指标](/zh-cn/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics)。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 17]
```diff
@@ -0,0 +1,242 @@
...
+
+<!--
+### CEL for admission control ([KEP-3488](https://kep.k8s.io/3488))
+-->
+### CEL 用于准入控制 ([KEP-3488](https://kep.k8s.io/3488))
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:211]:
```suggestion
### 在准入控制中使用 CEL ([KEP-3488](https://kep.k8s.io/3488))
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 18]
```diff
@@ -0,0 +1,242 @@
...
+
+<!--
+Integrating Common Expression Language (CEL) for admission control in Kubernetes introduces a more
+dynamic and expressive way of evaluating admission requests. This feature allows complex,
+fine-grained policies to be defined and enforced directly through the Kubernetes API, enhancing
+security and governance capabilities without compromising performance or flexibility.
+-->
+在 Kubernetes 中，为准入控制集成了 Common Expression Language (CEL) 。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:219]:
```suggestion
Kubernetes 为准入控制集成了 Common Expression Language (CEL) 。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 19]
```diff
@@ -0,0 +1,242 @@
...
+这引入了一种更动态和更富表现力的评估准入请求的方式。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:220]:
```suggestion
这一集成引入了一种更动态、表达能力更强的方式来判定准入请求。
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 20]
```diff
@@ -0,0 +1,242 @@
...
+CEL's addition to Kubernetes admission control empowers cluster administrators to craft intricate
+rules that can evaluate the content of API requests against the desired state and policies of the
+cluster without resorting to Webhook-based access controllers. This level of control is crucial for
+maintaining the integrity, security, and efficiency of cluster operations, making Kubernetes
+environments more robust and adaptable to various use cases and requirements. For more information
+on using CEL for admission control, see the [API
+documentation](/docs/reference/access-authn-authz/validating-admission-policy/) for
+ValidatingAdmissionPolicy.
+-->
+CEL 的加入使 Kubernetes 的准入控制赋予了集群管理员制定复杂规则的能力，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:233]:
```suggestion
将 CEL 引入到 Kubernetes 的准入控制后，集群管理员就具有了制定复杂规则的能力，
```

File: content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md [hunk 21]
```diff
@@ -0,0 +1,242 @@
...
+这些规则可以根据集群的期望状态和策略来评估 API 请求的内容，而无需使用基于 Webhook 的访问控制器。
+这种控制水平对于维护集群操作的完整性、安全性和效率至关重要，使 Kubernetes 环境更加健壮，更适应各种用例和需求。
+有关使用 CEL 进行准入控制的更多信息，请参阅 [API 文档](/zh-cn/docs/reference/access-authn-authz/validating-admission-policy/)中的 ValidatingAdmissionPolicy。
+
+<!--
+We hope you're as excited for this release as we are. Keep an eye out for the official release 
+blog in a few weeks for more highlights!
+-->
+我们希望你和我们一样对这个版本的发布感到兴奋。请在未来几周内密切关注官方发布博客，以获取更多亮点！
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-03-12-mid-cycle-1.30.md:242]:
```suggestion
我们希望你和我们一样对这个版本的发布感到兴奋。请在未来几周内密切关注官方发布博客，以了解其他亮点！
```

**Comment by pacoxu** [comment]
@tengqm  thanks for your detailed review. All comments are addressed except the date.

This is the v1.30 release mid cycle blog which is directly released on March 12th in English version.
Do we need to change to a new date for Chinese version?

What date do you prefer? Is 03-17 OK?

**Comment by tengqm** [comment]
Date should be okay.
/lgtm
/approve

**Pull request merged**